src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -72,7 +72,7 @@ static std::string attributeType
     /* It could happen that different entities within the same entity type may have attributes with the same name
      * but different types. In that case, one type (at random) is returned. A list could be returned but the
      * NGSIv2 operations only allow to set one type */
-    return r.getField(ENT_ATTRS).embeddedObject().getField(attrName).embeddedObject().getStringField(ENT_ATTRS_TYPE);
+    return getStringField(getField(getField(r, ENT_ATTRS).embeddedObject(), attrName).embeddedObject(), ENT_ATTRS_TYPE);
```

Comment:
Pretty complex line. [ and yes, I know it is not new to this PR, it was just as bad before :-) ]
I would propose to split it up in 3 lines, using help variables (don't know what name to use for these).

Something like:

    XXX xObj = getField(r, ENT_ATTRS).embeddedObject());
    XXX yObj = getField(xObj, attrName).embeddedObject());

    return getStringField(yObj, ENT_ATTRS_TYPE);